### ISSUE TITLE
What do I do if my program is too big and will be getting bigger?

### ISSUE THREAD

**Issue by firestar115** [CE-Programming/toolchain#52]
[firestar115/TIWaifu](https://github.com/firestar115/TIWaifu)

I don't know why it is too big (probably because of my image size), but how in the future (with the changes mentioned in the `readme`) can I keep it at an acceptable size? Currently the calculator runs out of RAM.

My goal is to recreate my BASIC program but without the small size and refresh delay.

[WAIFU.zip](https://github.com/CE-Programming/toolchain/files/819431/WAIFU.zip)

(CEmu was acting up and wouldn't give me my GIF)

**Comment by adriweb** [comment]
Try sprite compression (and program compression, although that only postpones the problem at runtime)

**Comment by firestar115** [comment]
This is usually the outcome of running the program (other times it just fills the screen with random colors in the color palette)

![capture](https://cloud.githubusercontent.com/assets/7751296/23588756/04e133f8-0177-11e7-816b-eec342097c11.gif)

I am using the `COMPRESSION` option in the `Makefile`, what's sprite compression?

**Comment by firestar115** [comment]
After using sprite compression, everything is labeled with `_compressed`. When trying to `make` I get:

    C:\USERS\APRIM\CLIONPROJECTS\WAIFU\SRC\MAIN.C   (27,32) :       ERROR (171) Argument type is not compatible with formal parameter
    C:\USERS\APRIM\CLIONPROJECTS\WAIFU\SRC\MAIN.C   (28,33) :       ERROR (171) Argument type is not compatible with formal parameter

**Comment by MattWaltz** [comment]
I have released a new version of the toolchain which fixes the weird bug that you were experiencing when trying to run your program. Feel free to grab it here:

https://github.com/CE-Programming/toolchain/releases/latest

However, program compression probably will not cut it for you as these sprites are quite large (You won't be able to even run the program without storing it in the archive and using Cesium). So, I also wrote a wiki page to help you understand sprite compression, available here:

https://github.com/CE-Programming/toolchain/wiki/Sprite-Compression

Hopefully this is helps! Good luck on your project :)
